### PR TITLE
Pmanoj/upgrade to comp 0.0.10

### PR DESCRIPTION
**Pull request by jpmann**
Changes

Upgraded environment version from 12 -> 13
Upgraded component (both command and pipeline) versions from 9 -> 10
Removed gradient checkpointing
ignore mismatched sizes changed from model_name -> model_type
Removed test_data_mlpath from evaluation components

### CODE REVIEW

File: assets/training/finetune_acft_hf_nlp/src/finetune/finetune.py [hunk 1]
```diff
@@ -401,7 +424,35 @@ def finetune(args: Namespace):
...
+            # pass base model signature if available
+            mlflow_model_signature = mlflow_data.get("signature", {})
+            args.mlflow_ft_conf["mlflow_model_signature"].update(mlflow_model_signature)
+            logger.info(f"Updating signature from base model - {mlflow_model_signature}")
+        else:
+            logger.info("MLmodel file is empty")
+    else:
+        logger.info("MLmodel file does not exist")
+
+    logger.info(f"FT MLFlow config - {args.mlflow_ft_conf}")
```
Comment (skanakamedal) [inline assets/training/finetune_acft_hf_nlp/src/finetune/finetune.py:455]: remove this as base model signatures are for fill mask task

File: assets/training/finetune_acft_hf_nlp/src/finetune/finetune.py [hunk 2]
```diff
@@ -415,11 +466,6 @@ def finetune(args: Namespace):
...
         args.deepspeed = None
 
-    # enable gradient checkpointing
-    if hasattr(args, "model_name") and args.model_name in ENABLE_GRADIENT_CHECKPOINTING:
-        logger.info(f"Enabling gradient checkpointing for {args.model_name}")
-        setattr(args, "gradient_checkpointing", True)
-
     if args.deepspeed:
         with open(args.deepspeed, "r") as fp:
             ds_data = json.load(fp)
```
Comment (skanakamedal) [inline assets/training/finetune_acft_hf_nlp/src/finetune/finetune.py:471]: also remove https://github.com/Azure/azureml-assets/blob/0e2bb0bb6038518bc1b66006ba6b19f56356a64a/assets/training/finetune_acft_hf_nlp/src/finetune/finetune.py#L509C1-L549C1

**Review by skanakamedal — CHANGES_REQUESTED** [review]

File: assets/training/finetune_acft_hf_nlp/src/finetune/finetune.py (hunk 1 above)
Comment (jpmann) [reply assets/training/finetune_acft_hf_nlp/src/finetune/finetune.py]: removed the dependency on the base model signature

**Review by skanakamedal — APPROVED** [review]

File: assets/training/finetune_acft_hf_nlp/src/finetune/finetune.py [hunk 3]
```diff
@@ -385,10 +396,21 @@ def finetune(args: Namespace):
...
-        logger.info(f"Forcing `ignore_mismatched_sizes` to False for {args.model_name}")
+    if hasattr(args, "model_type") and args.model_type in IGNORE_MISMATCHED_SIZES_FALSE_MODELS:
+        logger.info(f"Identified model type: {args.model_type}. Forcing `ignore_mismatched_sizes` to False.")
         setattr(args, "ignore_mismatched_sizes", False)
 
+    # set `mlflow_ft_conf` - contains all mlflow related properties
+    setattr(args, "mlflow_ft_conf", {})
+
+    # set task based mlflow_model_signature
+    if getattr(args, "task_name", None) is not None and args.task_name in MLFLOW_MODEL_SIGNATURES:
```
Comment (ghyadav) [inline assets/training/finetune_acft_hf_nlp/src/finetune/finetune.py:407]: Are we adding signatures? We will not be able to the schema with parameters for online deployment then

**Review by sarthaks95 — APPROVED** [review]

**Review by skasturi — APPROVED** [review]

**Comment by skasturi** [comment]
As discussed, please add UTs & a workflow that run them.
